### PR TITLE
[EC-205] personal_sign & eth_sign methods

### PR DESCRIPTION
**Pull request by AlanVerbner**
## Description

As `personal_sign` and `eth_sign` methods are almost the same, this PR includes both. The only difference is:

- `eth_sign` requires the account to be unlocked when calling the method
- `personal_sign` unlocks the account but just for that call

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala [hunk 1]
```diff
@@ -401,4 +401,17 @@ object EthJsonMethodsImplicits extends JsonMethodsImplicits {
     def encodeJson(t: GetTransactionCountResponse): JValue = encodeAsHex(t.value)
   }
 
+  implicit val eth_sign = new JsonDecoder[SignRequest] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, SignRequest] =
+      params match {
+        case Some(JArray(JString(message) :: JString(addr) :: _)) =>
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala]: Maybe the order of the parameters should be `[addr, message]`? It seems to be that way in https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign.

Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala]: good catch...let me check :cry:

**Comment by ntallar** [comment]
I'm having an issue when testing this that might not have been caused by the changes on this PR (so could be changed in a separate task if needed). 
The following request works on our client but doesn't work in other clients (apart from the order of the parameters):
`"jsonrpc":"2.0","method":"eth_sign","params": ["0x2565023b2980f668ef86b11a27e628e5ff696235", "1234"],"id":1`
This seems to be caused by other clients checking whether `DATA` parameters has a `0x` prefix or not

**Comment by ntallar** [comment]
Apart from those comments, LGTM!

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala (hunk 1 above)
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala]: Done!

**Comment by AlanVerbner** [comment]
@ntallar fixed!

**Pull request merged**